daemon/cmd/daemon_main.go

```diff
@@ -1478,20 +1480,17 @@ func initEnv(cmd *cobra.Command) {
 		if !option.Config.EnableIPv4 {
 			option.Config.EnableIPv4FragmentsTracking = false
 		} else {
-			supportedMapTypes := probes.NewProbeManager().GetMapTypes()
-			if !supportedMapTypes.HaveLruHashMapType {
+			if probes.HaveMapType(ebpf.LRUHash) != nil {
 				option.Config.EnableIPv4FragmentsTracking = false
 				log.Info("Disabled support for IPv4 fragments due to missing kernel support for BPF LRU maps")
 			}
 		}
 	}
 
 	if option.Config.EnableBPFTProxy {
-		if h := probes.NewProbeManager().GetHelpers("sched_act"); h != nil {
-			if _, ok := h["bpf_sk_assign"]; !ok {
-				option.Config.EnableBPFTProxy = false
-				log.Info("Disabled support for BPF TProxy due to missing kernel support for socket assign (Linux 5.7 or later)")
-			}
+		if probes.HaveProgramHelper(ebpf.SchedCLS, asm.FnSkAssign) != nil {
```

Comment:
The previous code says `sched_act`, but the new code says `SchedCLS`. I don't think these are the same thing?

Reply:
(ditto for the rest of this commit, I won't repeat the comment in each location)

Reply:
Hmm, the more I look through the PR, the more it seems like the existing code was just inconsistently checking `sched_act` vs. `sched_cls`. From memory I think we actually use `sched_act` (direct action) for attachment rather than attaching a classifier, but I could be wrong.

Reply:
`sched_act` and `sched_cls` were introduced in the same commit and have been in the kernel almost since the start of ebpf. `sk_assign` can be used in both cls and act, so the probe result here should be equivalent. Not sure why this was changed, but at least now we're consistently using `SchedCLS` as you pointed out! :wink: